### PR TITLE
[Merged by Bors] - feat: relationship between Mellin transform/inverse and Fourier transform/inverse

### PR DESCRIPTION
**Pull request by llllvvuu**
---

- [x] depends on: #10810

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/MellinInversion.lean [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    rw [MellinConvergent, ← rexp_neg_image_aux, integrableOn_image_iff_integrableOn_abs_deriv_smul
+      MeasurableSet.univ rexp_neg_deriv_aux rexp_neg_injOn_aux] at hf
+    replace hf : Integrable fun (x : ℝ) ↦ cexp (-↑σ * ↑x) • f (rexp (-x)) := by
+      simpa [rexp_cexp_aux] using hf
+    norm_cast at hf
+  replace hFf : Integrable (𝓕 g) := by
+    have h2π : 2 * π ≠ 0 := by norm_num; exact pi_ne_zero
+    haveI : HasDistribNeg ℝ := inferInstance
+    simpa [VerticalIntegrable, mellin_eq_fourierIntegral, mul_div_cancel _ h2π] using
+      hFf.comp_mul_right' h2π
```
Comment (sgouezel) [inline Mathlib/Analysis/MellinInversion.lean:100]:
```suggestion
    simpa only [ofReal_mul, ofReal_ofNat, mellin_eq_fourierIntegral, add_re, ofReal_re, mul_re,
      re_ofNat, im_ofNat, ofReal_im, mul_zero, sub_zero, mul_im, zero_mul, add_zero, I_re, I_im,
      mul_one, sub_self, add_im, zero_add, mul_div_cancel _ h2π] using hFf.comp_mul_right' h2π
```
There is something fishy going on here with the instance `HasDistribNeg ℝ`. The `simpa only` is probably more robust.

File: Mathlib/Analysis/MellinInversion.lean [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  replace hfx : ContinuousAt g (-Real.log x) := by
+    refine ContinuousAt.smul (by fun_prop) (ContinuousAt.comp ?_ (by fun_prop))
+    simpa [Real.exp_log hx] using hfx
+  calc
+    _ = mellin_inv σ (fun s ↦ 𝓕 g (s.im / (2 * π))) x := by
```
Comment (sgouezel) [inline Mathlib/Analysis/MellinInversion.lean:105]: Can you spell out the first and last terms in calc blocks? (i.e., don't start or end with an underscore). Lean doesn't need it, but it helps readers a lot.

File: Mathlib/Analysis/MellinTransform.lean [hunk 3]
```diff
@@ -99,12 +99,21 @@ theorem MellinConvergent.comp_rpow {f : ℝ → E} {s : ℂ} {a : ℝ} (ha : a 
...
+
 /-- The Mellin transform of a function `f` (for a complex exponent `s`), defined as the integral of
 `t ^ (s - 1) • f` over `Ioi 0`. -/
 def mellin (f : ℝ → E) (s : ℂ) : E :=
   ∫ t : ℝ in Ioi 0, (t : ℂ) ^ (s - 1) • f t
 #align mellin mellin
 
+/-- The Mellin inverse transform of a function `f`, defined as `1 / (2π)` times
+the integral of `y ↦ x ^ -(σ + yi) • f (σ + yi)`. -/
+def mellin_inv (σ : ℝ) (f : ℂ → E) (x : ℝ) : E :=
```
Comment (sgouezel) [inline Mathlib/Analysis/MellinTransform.lean:114]:
```suggestion
def mellinInv (σ : ℝ) (f : ℂ → E) (x : ℝ) : E :=
```
It's one single thing, so no underscore in it. Can you please also fix the various places where it's used?

**Review by sgouezel — COMMENTED** [review]
Looks good, thanks!

File: Mathlib/Analysis/MellinInversion.lean (hunk 1 above)
Comment (llllvvuu) [reply Mathlib/Analysis/MellinInversion.lean]: I was having the `simpa only` succeed locally and fail in CI. I've added an intermediate goal which should hopefully be more robust.

File: Mathlib/Analysis/MellinInversion.lean [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+-/
+import Mathlib.Analysis.Fourier.Inversion
+
+/-!
+# Mellin inversion formula
+
+We derive the Mellin inversion formula as a consequence of the Fourier inversion formula.
+
+## Main results
+- `mellinInversion`: The inverse Mellin transform of the Mellin transform applied to `x > 0` is x.
```
Comment (sgouezel) [inline Mathlib/Analysis/MellinInversion.lean:14]:
```suggestion
- `mellin_inversion`: The inverse Mellin transform of the Mellin transform applied to `x > 0` is x.
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Pull request closed without merging**